lang/typescript/src/utils/address-fields.test.ts

```diff
@@ -214,4 +215,93 @@ describe('splitAddressField', () => {
       });
     });
   });
+
+  describe('regexSplitAddressField', () => {
+    test('creates an address object from string matching one of the defined regexes', () => {
+      const fieldDefinition: FieldConcatenationRule[] = [
+        {key: 'streetName'},
+        {key: 'streetNumber'},
+      ];
+      const regexPatterns = [
+        new RegExp('^(?<streetName>[^\\d]+) (?<streetNumber>\\d+)$'),
+      ];
```

Comment:
nit: can you update this test to have 2 regexPatterns, with the first not a match and the second a match?